suite/tests/api/ir_riscv64.c

```diff
@@ -1426,6 +1426,159 @@ test_misc(void *dc)
     test_instr_encoding(dc, OP_cbo_inval, instr);
 }
 
+static void
+test_xinst(void *dc)
+{
+    instr_t *instr;
+    byte *pc;
+
+    instr =
+        XINST_CREATE_load(dc, opnd_create_reg(DR_REG_A0),
+                          opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_4));
+    pc = test_instr_encoding(dc, OP_lwu, instr);
+    instr =
+        XINST_CREATE_load(dc, opnd_create_reg(DR_REG_A0),
+                          opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_8));
+    test_instr_encoding(dc, OP_ld, instr);
+
+    instr = XINST_CREATE_load_1byte_zext4(
+        dc, opnd_create_reg(DR_REG_A0),
+        opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_1));
+    test_instr_encoding(dc, OP_lbu, instr);
+
+    instr = XINST_CREATE_load_1byte(
+        dc, opnd_create_reg(DR_REG_A0),
+        opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_1));
+    test_instr_encoding(dc, OP_lbu, instr);
+
+    instr = XINST_CREATE_load_2bytes(
+        dc, opnd_create_reg(DR_REG_A0),
+        opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_2));
+    test_instr_encoding(dc, OP_lhu, instr);
+
+    instr = XINST_CREATE_store(
+        dc, opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_4),
+        opnd_create_reg(DR_REG_A0));
+    test_instr_encoding(dc, OP_sw, instr);
+
+    instr = XINST_CREATE_store(
+        dc, opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_8),
+        opnd_create_reg(DR_REG_A0));
+    test_instr_encoding(dc, OP_sd, instr);
+
+    instr = XINST_CREATE_store_1byte(
+        dc, opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_1),
+        opnd_create_reg(DR_REG_A0));
+    test_instr_encoding(dc, OP_sb, instr);
+
+    instr = XINST_CREATE_store_2bytes(
+        dc, opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_2),
+        opnd_create_reg(DR_REG_A0));
+    test_instr_encoding(dc, OP_sh, instr);
+
+    instr = XINST_CREATE_move(dc, opnd_create_reg(DR_REG_A0), opnd_create_reg(DR_REG_A1));
+    ASSERT(opnd_is_immed_int(instr_get_src(instr, 1)) &&
+           opnd_get_immed_int(instr_get_src(instr, 1)) == 0);
+    test_instr_encoding(dc, OP_addi, instr);
+
+    /* FIXME: i#3544 Currently, these two exist as placeholders on RISCV64:
```

Comment:
So the simd ones indeed still need some work? Maybe add a FIXME at their definition too.

Reply:
> So the simd ones indeed still need some work?

Explained above.

> Maybe add a FIXME at their definition too.

Done.